osu.Game.Rulesets.Osu/Objects/Drawables/DrawableSpinner.cs

```diff
@@ -221,8 +225,7 @@ protected override void Update()
                 RotationTracker.Tracking = !Result.HasResult && (OsuActionInputManager?.PressedActions.Any(x => x == OsuAction.LeftButton || x == OsuAction.RightButton) ?? false);
 
             if (spinningSample != null)
-                // todo: implement SpinnerFrequencyModulate
-                spinningSample.Frequency.Value = 0.5f + Progress;
+                spinningSample.Frequency.Value = spinnerFrequencyModulate ? 0.5f + Progress : 0.5f;
```

Comment:
move the initial set of frequency to 0.5f to the point the sample is constructed, then

```csharp
if (spinningSample != null && spinnerFrequencyModulate)
    spinningSample.Frequency.Value = 0.5f + Progress;
```

also move 0.5f to a constant.